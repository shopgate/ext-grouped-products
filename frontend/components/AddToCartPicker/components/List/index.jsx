import React, { Component } from 'react';
import PropTypes from 'prop-types';
import BaseList from '@shopgate/pwa-common/components/List';
import BaseListItem from '@shopgate/pwa-common/components/List/components/Item';
import Item from './components/Item';
import styles from './style';

/**
 * The List component.
 */
class List extends Component {
  static Item = Item;

  static propTypes = {
    children: PropTypes.node,
  };

  static defaultProps = {
    children: null,
  };

  /**
   * Renders the component.
   * @returns {JSX}
   */
  render() {
    const { children } = this.props;

    if (!React.Children.count(children)) {
      return null;
    }

    return (
      <BaseList>
        {React.Children.map(children, (child, index) => {
          if (!React.isValidElement(child)) {
            return null;
          }
          // The key for each child.
          const key = `child-${index}`;
          // Selected state for the child.
          const { isSelected } = child.props;

          const classes = styles.item;

          return (
            <BaseListItem
              className={classes}
              isSelected={isSelected}
              key={key}
            >
              <div className={styles.innerContainer}>
                {child}
              </div>
            </BaseListItem>
          );
        })}
      </BaseList>
    );
  }
}

export default List;
