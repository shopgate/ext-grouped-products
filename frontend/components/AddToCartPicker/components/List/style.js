import { css } from 'glamor';
import { themeConfig } from '@shopgate/pwa-common/helpers/config';

const { colors } = themeConfig;

const item = css({
  marginLeft: 72,
}).toString();

const itemNotLast = css({
  boxShadow: `0 1px 0 0 ${colors.darkGray}`,
  marginBottom: 1,
}).toString();

const innerContainer = css({
  marginLeft: -72,
  minHeight: 56,
  position: 'relative',
}).toString();

const glow = css({
  bottom: -1,
  height: '100%',
  top: -1,
}).toString();

export default {
  item,
  itemNotLast,
  innerContainer,
  glow,
};
