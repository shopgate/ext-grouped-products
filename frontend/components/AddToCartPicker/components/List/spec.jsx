import React from 'react';
import { mount } from 'enzyme';
import List from './index';

describe('<List />', () => {
  it('should render with two children', () => {
    const wrapper = mount((
      <List>
        <List.Item title="List Item" />
        <List.Item title="List Item" />
      </List>
    ));

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(List.Item).length).toEqual(2);
  });
});
