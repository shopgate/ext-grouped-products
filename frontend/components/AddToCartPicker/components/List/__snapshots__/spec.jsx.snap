// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<List /> should render with two children 1`] = `
<List>
  <List
    className=""
  >
    <ul
      className="css-gkw1j5 "
    >
      <ListItem
        className="css-1brrprg css-1ym3l2n"
        isSelected={false}
        key="child-0/.0"
      >
        <li
          className="css-1p5q8vq css-1brrprg css-1ym3l2n css-16gj3o1"
        >
          <div
            className="css-1nnz83b"
          >
            <Item
              className={null}
              image={null}
              isDisabled={false}
              isSelected={false}
              link={null}
              onClick={null}
              rightComponent={null}
              testId={null}
              title="List Item"
            >
              <Grid
                className="css-1qcr7hz"
                component="div"
                wrap={false}
              >
                <div
                  className="css-1qcr7hz css-h7r55a"
                >
                  <div
                    className="css-jwxytc"
                  />
                  <GridItem
                    className="css-10fxwvh"
                    component="div"
                    grow={1}
                    shrink={1}
                  >
                    <div
                      className="css-10fxwvh css-1gs0ko2"
                    >
                      List Item
                    </div>
                  </GridItem>
                </div>
              </Grid>
            </Item>
          </div>
        </li>
      </ListItem>
      <ListItem
        className="css-1brrprg"
        isSelected={false}
        key="child-1/.1"
      >
        <li
          className="css-1p5q8vq css-1brrprg css-16gj3o1"
        >
          <div
            className="css-1nnz83b"
          >
            <Item
              className={null}
              image={null}
              isDisabled={false}
              isSelected={false}
              link={null}
              onClick={null}
              rightComponent={null}
              testId={null}
              title="List Item"
            >
              <Grid
                className="css-1qcr7hz"
                component="div"
                wrap={false}
              >
                <div
                  className="css-1qcr7hz css-h7r55a"
                >
                  <div
                    className="css-jwxytc"
                  />
                  <GridItem
                    className="css-10fxwvh"
                    component="div"
                    grow={1}
                    shrink={1}
                  >
                    <div
                      className="css-10fxwvh css-1gs0ko2"
                    >
                      List Item
                    </div>
                  </GridItem>
                </div>
              </Grid>
            </Item>
          </div>
        </li>
      </ListItem>
    </ul>
  </List>
</List>
`;
